anvil-python/anvil/provider/local/commands.py

```diff
@@ -462,6 +465,8 @@ def join(
             )
         )
 
+    plan2.append(JujuAddSshKeyStep())
```

Comment:
IMHO this has to be the last step of `plan1` because this is the place where you are adding the machine to juju model.